ui/page/odyseeMembership/view.jsx

```diff
@@ -0,0 +1,462 @@
+    e.preventDefault();
+    e.stopPropagation();
+
+    console.log(membershipOption);
+    console.log(price);
+
+    const planName = membershipOption.Membership.name;
+
+    const membershipId = e.currentTarget.getAttribute('membership-id');
+    const priceId = e.currentTarget.getAttribute('price-id');
+    const purchaseString = buildPurchaseString(price.unit_amount, price.recurring.interval, planName);
+
+    openModal(MODALS.CONFIRM_ODYSEE_MEMBERSHIP, {
+      membershipId,
+      odyseeChannelId: userChannelClaimId, // TODO: this needs to be renamed
+      odyseeChannelName: userChannelName,
+      priceId,
+      purchaseString,
+      plan: planName,
+    });
+  };
+
+  const cancelMembership = async function (e, membership) {
+    const membershipId = e.currentTarget.getAttribute('membership-id');
+
+    console.log(membership);
+
+    const cancellationString = 'You are cancelling your Odysee Membership. You will still have access to all the paid ' +
+      'features until the point of the expiration of your current membership, at which point you will not be charged ' +
+      'again and your membership will no longer be active.';
+
+    openModal(MODALS.CONFIRM_ODYSEE_MEMBERSHIP, {
+      membershipId,
+      hasMembership,
+      purchaseString: cancellationString,
+    });
+  };
+
+  function convertPriceToString(price) {
+    const interval = price.recurring.interval;
+
+    if (interval === 'year') {
+      return 'Yearly';
+    } else if (interval === 'month') {
+      return 'Monthly';
+    }
+  }
+
+  function capitalizeWord(string) {
+    return string.charAt(0).toUpperCase() + string.slice(1);
+  }
+
+  function buildCurrencyDisplay(priceObject) {
+    let currencySymbol;
+    if (priceObject.currency === 'eur') {
+      currencySymbol = '€';
+    } else if (priceObject.currency === 'usd') {
+      currencySymbol = '$';
+    }
+
+    const currency = priceObject.currency.toUpperCase();
+
+    return currency + ' ' + currencySymbol;
+  }
+
+  const urlSearchParams = new URLSearchParams(window.location.search);
+  const params = Object.fromEntries(urlSearchParams.entries());
+
+  const { interval, plan, pageLocation } = params;
+
+  console.log(params);
+  const confirmValue = params.confirm;
+  const planValue = params.plan;
+  // const pageLocation = params.pageLocation;
+
+  // add a bit of a delay otherwise it's a bit jarring
+  let timeoutValue = 300;
+  if (pageLocation === 'confirmPage') {
+    timeoutValue = 300;
+  }
+
+  if (!stillWaitingFromBackend && planValue) {
+    setTimeout(function() {
+      // clear query params
+      window.history.replaceState(null, null, window.location.pathname);
+
+      setHasShownModal(true);
+
+      // open confirm purchase modal
+      // document.querySelector(`[plan=${plan}][interval=${interval}]`).click();
+      document.querySelector('[plan="' + plan + '"][interval="' + interval + '"]').click();
+    }, timeoutValue);
+  }
+
```

Comment:
Render functions must be "pure". 
`setTimeout` is usually called in a hook (useEffect) or event handlers.

Is this just trying to programatically click a button?  If yes, there are React ways to do it.